src/model/index.py

```diff
@@ -0,0 +1,19 @@
+import os
+from output.exceptions import *
+from commands.keys import Keys
+
+
+class Index:
+    def __init__(self, name, table_metadata, path):
+        self.name = name
+        self.table_metadata = table_metadata
+        self.path = os.path.join(path, self.name)
+        self.__validate__(self)
+
+    @staticmethod
+    def __validate__(self):
```

Comment:
How is ```__validate``` ```static``` and it takes self? 
The ```@staticmethod``` doesn’t take instance or class as a parameter because they don’t have access to the instance variables and class variables.
It shouldn't be static

Reply:
I recommend reading this site. It helped me a lot in understanding the difference
https://pynative.com/python-class-method-vs-static-method-vs-instance-method/#:~:text=The%20instance%20method%20acts%20on,class%20attributes%20(class%20variables).

Reply:
+1

Reply:
Thanks for the information.